documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,633 @@
+
+#### Flow encap related attributes
+
+These are the related attributes of flow encapsulation.
+
+| Attribute name                                   | Type                       | Description                                                  |
+| ------------------------------------------------ | -------------------------- | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_VNI                | `sai_uint32_t`             | Destination VNI in the underlay network                      |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_SIP                | `sai_uint32_t`             | Source IP address in the underlay network                    |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DIP                | `sai_uint32_t`             | Destination IP address in the underlay network               |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_SMAC               | `sai_mac_t`                | Source MAC address in the underlay network                   |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DMAC               | `sai_mac_t`                | Destination MAC address in the underlay network              |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY0_DASH_ENCAPSULATION | `sai_dash_encapsulation_t` | Encapsulation method for DASH traffic in the underlay network |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_VNI                | `sai_uint32_t`             | Destination VNI in the 2nd underlay network                  |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_SIP                | `sai_uint32_t`             | Source IP address in the 2nd underlay network                |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DIP                | `sai_uint32_t`             | Destination IP address in the 2nd underlay network           |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_SMAC               | `sai_mac_t`                | Source MAC address in the 2nd underlay network               |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DMAC               | `sai_mac_t`                | Destination MAC address in the 2nd underlay network          |
+| SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DASH_ENCAPSULATION | `sai_dash_encapsulation_t` | Encapsulation method for DASH traffic in the 2nd underlay network |
+
+#### Flow overlay rewrite related attributes
+
+These are the related attributes of flow rewrite.
+
+| Attribute name               | Type               | Description                                                  |
+| ---------------------------- | ------------------ | ------------------------------------------------------------ |
+| SAI_FLOW_ENTRY_ATTR_DST_MAC  | `sai_mac_t`        | Destination MAC address for the flow entry.                  |
+| SAI_FLOW_ENTRY_ATTR_SIP      | `sai_ip_address_t` | Source IP address for the flow entry, supporting both IPv4 and IPv6. |
+| SAI_FLOW_ENTRY_ATTR_DIP      | `sai_ip_address_t` | Destination IP address for the flow entry, supporting both IPv4 and IPv6. |
+| SAI_FLOW_ENTRY_ATTR_SIP_MASK | `sai_ip_address_t` | Subnet mask for the source IP address.                       |
```

Comment:
i am a bit hesitated on this one. saving prefix can indeed help saving some memory in certain cases, for example, service tunnel. but on the other hand, sip and dip might also be good enough...

Reply:
Synced offline. Let's keep it.